tests/unit/testplan/web_ui/testing/test_jest.py

```diff
@@ -46,4 +46,4 @@ def test_testplan_ui():
 )
 def test_eslint():
     """Run eslint over the UI source code."""
-    subprocess.check_call(["npm", "run", "lint"], cwd=TESTPLAN_UI_DIR)
```

Comment:
I wonder why is this change?

Reply:
Without `shell=True` on my windows machine the testcase does not pick up PATH and do not found `npm`, however when `shell=True` the behaviour of the popen changes a bit, best describe in [this article](https://medium.com/python-pandemonium/a-trap-of-shell-true-in-the-subprocess-module-6db7fc66cdfd).  This looks to be a sweet spot for windows and linux, as well as it is in sync with the test_testplan_ui method just above.

Reply:
I saw the same error running on Windows. I didn't think to change it because it is just as easy to run the linter directly as `npm run lint`; this pytest wrapper is mostly here just to ensure that the linter is run as part of the Travis build.

It looks like the error occurs because, while on Linux `npm` is the name of the executable, on Windows the executable is `npm.CMD` and `npm` is just a kind of shell command for convenience. So when not setting shell=True, python cannot find the `npm` executable (because it doesn't exist). I found a good SO question which explains it better here: https://stackoverflow.com/questions/3022013/windows-cant-find-the-file-on-subprocess-call/43580381